### PR TITLE
Ignore case when parsing units

### LINKED ISSUE #17: Support lowercase `kiB`
Currently, `Size::from_str` only supports uppercase units: `KiB`, `KB`. It would be nice to support having the first letter be lowercased, which doesn't change the meaning in any human-readable format I'm aware of.

### PR DESCRIPTION
**Pull request by jyn514**
Fixes https://github.com/Thomasdezeeuw/human-size-rs/issues/17.

### CODE REVIEW

File: src/multiples.rs [hunk 1]
```diff
@@ -162,10 +162,15 @@ impl FromStr for Any {
     type Err = ParsingError;
 
     fn from_str(input: &str) -> Result<Any, Self::Err> {
-        match input {
+        // Special case this to mean kibibytes.
+        // FIXME(#10): this is not ideal.
+        if input == "KB" {
+            return Ok(Any::Kibibyte);
+        }
+        match &*input.to_uppercase() {
```
Comment (Thomasdezeeuw) [inline src/multiples.rs]: This allocate a new string just for this match. Can you use something like `"b" | "B"` instead, that would avoid the allocation.

Comment (jyn514) [reply src/multiples.rs]: I used `eq_ignore_ascii_case` :) to avoid having to quadruple the number of match patterns.

**Comment by Thomasdezeeuw** [comment]
Thanks @jyn514!

**Comment by jyn514** [comment]
Thank you!

**Pull request merged**